core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -194,6 +197,64 @@ class AdminManager(val config: KafkaConfig,
     }
   }
 
+  def createPartitions(timeout: Int,
+                       newPartitions: Map[String, NewPartitions],
+                       validateOnly: Boolean,
+                       listenerName: ListenerName,
+                       callback: Map[String, ApiError] => Unit): Unit = {
+    var waiting = Map[String, NewPartitions]()
+    var done = Map[String, ApiError]()
+
+    newPartitions.foreach{case (topic, newPartition: NewPartitions) =>
+      try {
+        val oldNumPartitions = zkUtils.getTopicPartitionCount(topic) match {
+          case Some(count) => count
+          case None => throw new InvalidTopicException()
+        }
+        val newNumPartitions = newPartition.totalCount()
+        val numPartitionsIncrement = newNumPartitions - oldNumPartitions
+        if (numPartitionsIncrement <= 0) {
+          throw new InvalidPartitionsException(s"Topic currently has $oldNumPartitions partitions: Cannot create $numPartitionsIncrement partitions.")
+        }
+        val reassignmentStr = if (newPartition.assignments() == null) {
```

Comment:
You can use `match` instead of `if`:
`newPartition.assignments match {
case null => ""
case assignments => ...
}

Reply:
In cases like this, either option is fine. I even prefer `if/else` since it's more concise. Pattern matching is beneficial when it's used to extract data, but that is not happening here.

Reply:
@ijuma is the scala expert, so please leave as is :-)